src/dbnode/persist/fs/retriever.go

```diff
@@ -297,73 +296,70 @@ func (r *blockRetriever) fetchLoop(seekerMgr DataFileSetSeekerManager) {
 	r.fetchLoopsHaveShutdownCh <- struct{}{}
 }
 
-func (r *blockRetriever) processIndexChecksumRequest(
-	req *retrieveRequest,
-	seeker ConcurrentDataFileSetSeeker,
-	seekerResources ReusableSeekerResources,
-) {
-	checksum, err := seeker.SeekIndexEntryToIndexChecksum(req.id, seekerResources)
-	if err != nil {
-		req.onError(err)
-		return
-	}
-
-	req.onIndexChecksumCompleted(checksum)
-	req.onCallerOrRetrieverDone()
-}
-
-func (r *blockRetriever) processReadMismatchRequest(
-	req *retrieveRequest,
-	seeker ConcurrentDataFileSetSeeker,
-	seekerResources ReusableSeekerResources,
-) {
-	checksum, err := seeker.SeekIndexEntryToIndexChecksum(req.id, seekerResources)
-	if err != nil {
-		req.onError(err)
-		return
-	}
-
-	mismatch, err := seeker.SeekReadMismatchesByIndexChecksum(
-		checksum, req.mismatchChecker, seekerResources)
-
-	if err != nil && err != errSeekIDNotFound {
-		req.onError(err)
-		return
-	}
-
-	if err == errSeekIDNotFound {
-		req.onIndexMismatchCompleted(wide.ReadMismatch{})
-		return
-	}
-
-	req.onIndexMismatchCompleted(mismatch)
-	req.onCallerOrRetrieverDone()
-}
-
 // filterAndCompleteWideReqs completes all wide operation retrieve requests,
 // returning a list of requests that need to be processed by other means.
 func (r *blockRetriever) filterAndCompleteWideReqs(
 	reqs []*retrieveRequest,
 	seeker ConcurrentDataFileSetSeeker,
 	seekerResources ReusableSeekerResources,
+	retrieverResources *reuseableRetrieverResources,
 ) []*retrieveRequest {
-	filteredStreamRequests := reqs[:0]
+	retrieverResources.resetDataReqs()
+	retrieverResources.resetWideEntryReqs()
 	for _, req := range reqs {
 		switch req.streamReqType {
 		case streamDataReq:
 			// NB: filter out stream requests; these are handled outside of
 			// wide logic functions.
-			filteredStreamRequests = append(filteredStreamRequests, req)
-		case streamIdxChecksumReq:
-			r.processIndexChecksumRequest(req, seeker, seekerResources)
-		case streamReadMismatchReq:
-			r.processReadMismatchRequest(req, seeker, seekerResources)
+			retrieverResources.dataReqs = append(retrieverResources.dataReqs, req)
+
+		case streamWideEntryReq:
+			entry, err := seeker.SeekWideEntry(req.id, seekerResources)
+			if err != nil && err != errSeekIDNotFound {
+				req.err = err
+				continue
+			}
+
+			if err == errSeekIDNotFound {
+				// Missing, return empty result, successful lookup.
+				req.wideEntry = xio.WideEntry{}
```

Comment:
nit: maybe we should have `emptyWideEntry` method here just to be consistent with similar usage in `src/dbnode/persist/fs/msgpack/decoder.go`

Reply:
Yeah not too sure the best path here, typically I prefer to explicitly return an empty struct like `xio.WideEntry{}` here so that there's no chance of accidentally mutating `emptyWideEntry` itself, but went with the `emptyWideEntry` route in the decoder to match the standards there. We should probably try to converge to a single standard, but not sure what the best solution here would be, perhaps this can be added to a linter to make it easier to standardize on these? I think I may leave it as is for now, just so we don't need to expose the `EmptyWideEntry` struct from `xio` since there's no special behavior on it versus just empty `xio.WideEntry{}`